offerurl.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+package charm
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/juju/errors"
+	names "gopkg.in/juju/names.v2"
+)
+
+// OfferURL represents the location of an offered application and its
+// associated exported endpoints.
+type OfferURL struct {
```

Comment:
Does this mean that we can strip off some code from juju/juju and use this parser instead?

Reply:
@achilleasa exactly